diff --git a/src/cairo-ps-surface.c b/src/cairo-ps-surface.c
index 381b4cf..95aec4b 100644
--- a/src/cairo-ps-surface.c
+++ b/src/cairo-ps-surface.c
@@ -103,7 +103,7 @@
 #endif
 
 #ifndef HAVE_CTIME_R
-static char *ctime_r(const time_t *timep, char *buf)
+static char *ctime_r_s(const time_t *timep, char *buf)
 {
     (void)buf;
     return ctime(timep);
@@ -329,7 +329,7 @@ _cairo_ps_surface_emit_header (cairo_ps_surface_t *surface)
     if (!getenv ("CAIRO_DEBUG_PS_NO_DATE")) {
        _cairo_output_stream_printf (surface->final_stream,
                                     "%%%%CreationDate: %s",
-                                    ctime_r (&now, ctime_buf));
+                                    ctime_r_s (&now, ctime_buf));
     }
 
     _cairo_output_stream_printf (surface->final_stream,
