diff --git a/src/cairo-xlib-xrender-private.h b/src/cairo-xlib-xrender-private.h
index 9076946..6414337 100644
--- a/src/cairo-xlib-xrender-private.h
+++ b/src/cairo-xlib-xrender-private.h
@@ -102,34 +102,14 @@ __attribute__((__unused__)) static void   _void_consume_free   (Display *p, XID
 
 #if !HAVE_XRENDERCREATELINEARGRADIENT
 #define XRenderCreateLinearGradient                    _int_consume
-
-typedef struct _XLinearGradient {
-    XPointFixed p1;
-    XPointFixed p2;
-} XLinearGradient;
 #endif
 
 #if !HAVE_XRENDERCREATERADIALGRADIENT
 #define XRenderCreateRadialGradient                    _int_consume
-
-typedef struct _XCircle {
-    XFixed x;
-    XFixed y;
-    XFixed radius;
-} XCircle;
-typedef struct _XRadialGradient {
-    XCircle inner;
-    XCircle outer;
-} XRadialGradient;
 #endif
 
 #if !HAVE_XRENDERCREATECONICALGRADIENT
 #define XRenderCreateConicalGradient                   _int_consume
-
-typedef struct _XConicalGradient {
-    XPointFixed center;
-    XFixed angle; /* in degrees */
-} XConicalGradient;
 #endif
 
 
